common/cert.c

```diff
@@ -211,6 +211,36 @@ oe_result_t oe_get_crl_distribution_points(
     return result;
 }
 
+oe_result_t oe_cert_chain_get_root_cert(
+    const oe_cert_chain_t* chain,
+    oe_cert_t* cert)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    size_t length;
+
+    OE_CHECK(oe_cert_chain_get_length(chain, &length));
+    OE_CHECK(oe_cert_chain_get_cert(chain, length - 1, cert));
+    result = OE_OK;
+
+done:
+    return result;
+}
+
+oe_result_t oe_cert_chain_get_leaf_cert(
+    const oe_cert_chain_t* chain,
+    oe_cert_t* cert)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    size_t length;
+
+    OE_CHECK(oe_cert_chain_get_length(chain, &length));
```

Comment:
Could the call to oe_cert_chain_get_length be skipped?

Reply:
Yes, this call is redundant in the original implementation. I will remove it.